HousingManagementSystemApi/Gateways/AddressesCosmosGateway.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using HACT.Dtos;
+namespace HousingManagementSystemApi.Gateways
+{
+    using Helpers;
+
+    public class AddressesCosmosGateway : IAddressesGateway
+    {
+        private ICosmosAddressQueryHelper cosmosAddressQueryHelper;
+
+        public AddressesCosmosGateway(ICosmosAddressQueryHelper cosmosAddressQueryHelper)
+        {
+            this.cosmosAddressQueryHelper = cosmosAddressQueryHelper;
+        }
+
+        public async Task<IEnumerable<PropertyAddress>> SearchByPostcode(string postcode)
+        {
+            using var queryResultSetIterator = this.cosmosAddressQueryHelper.GetItemQueryIterator<PropertyAddress>(postcode);
+            var addresses = new List<PropertyAddress>();
+
+            while (queryResultSetIterator.HasMoreResults)
+            {
+                var currentResultSet = await queryResultSetIterator.ReadNextAsync();
+                foreach (var address in currentResultSet)
```

Comment:
Replace use of `foreach` loop with single method call, i.e.
```suggestion
                addresses.AddRange(currentResultSet);
```